### DISCUSSION TITLE
MTS200 , & HVAC mode in Auto

### DISCUSSION THREAD

**Discussion by john-wolff59** [krahabb/meross_lan#275] [Q&A]
Hello, 

I have an MTS200 thermostat.
Under Home assistant, I created an automation to switch the heating to 24 degrees depending on the conditions.
so I used:
service: climate.set_temperature
data:
     temperature: 24
target:
     device_id: xxxxxxxxxxxx

when I run manually or depending on the conditions, the set point temperature goes well at 24 degrees. it's works !

on the other hand, I would like to succeed in switching this thermostat back to automatic (depending on the programming of the thermostat) on other conditions.
But I can't do it, the following command is executed, but no way to switch to auto mode:

service: climate.set_hvac_mode
data:
     hvac_mode: auto
target:
     device_id: xxxxxxxxxxxx

in OFF mode, it works.

Does anyone have any idea to figure out what is wrong?
thanks

**Comment by krahabb** [comment]
Hello @john-wolff59,
the mts200 has different 'auto' modes which are not well represented by HA 'hvac_mode'
In meross_lan, I've decided to implement this modes through the 'preset modes' feature of HA (it seemed more logical to me based on how hvac_modes and preset_modes work in HA)
So, in meross_lan this is what happens:
when you use 'hvac_mode':
- 'Off' turns off the thermostat
- 'Heat' turns on the thermostat on the same mode it was when turned off (the mts200 has different heating modes ranging from 'eco', sleep', 'cool', 'schedule'....)
- 'Auto' should always set the mts200 in 'schedule' mode

If you want to explicitly set one of the mts200 working modes it would be better to use the `climate.set_preset_mode` service instead

Beside this detailed explanation of the behavior, I've checked the meross_lan implementation and I see a problem (like the one you're exposing) might arise when meross_lan 'thinks' the mts200 is already on when issuing the `climate.set_hvac_mode` from 'Heat' to 'Auto' and so doesn't issue a 'turn on' command. That might explain the issue.
Nevertheless, this kind of bug would also emerge by using the `climate.set_preset_mode`

You could 'bypass' the bug by explicitly calling 'climate.turn_on' in your automations if that fixes the issue I would happily patch the code so to always send the command

**Reply by john-wolff59** [reply]
Thank you very much for this quick return !
I'm not an expert, and I'm new to Home assistant.

When I use the '`hvac_mode`': auto; the command does nothing.
The MST200 does not switch to 'schedule' mode, regardless of the current mode on the MST200; off, manual...
it is the expected that I want, to switch to 'schedule' mode.

you write : 
"when you use 'hvac_mode':
'Auto' should always set the mts200 in 'schedule' mode"
=> in my case, it's not true :-(

I tried to use `climate.set_preset_mode`, I'm not sure what I did, but it doesn't seem to work either.

`Climate.turn_on` just turns MST200 on, it stays in the same mode as when it was turned off.

**Comment by john-wolff59** [comment]
@krahabb,
i have spend time yesterday to test `hvac_mode : auto` from the MTS200 in OFF, ON heating ON sleeping etc.
But the command :

service: climate.set_hvac_mode
data:
hvac_mode: auto
target:
device_id: xxxxxxxxxxxx

this does not override the MTS200 in 'schedule' mode.

**Comment by krahabb** [comment]
@john-wolff59,
at the moment, the only thing coming to mind is that the mapping between the mts200 native modes and the ones exposed in HA/meross_lan is wrong. 

In order to proceed to debugging I would need a ['diagnostic trace'](https://github.com/krahabb/meross_lan/wiki/Diagnostics#download-diagnostics) captured with the following conditions:
- put the mts200 in a 'well known' state (say: put it to 'manual mode' and then turn it off)
- start the diagnostics.
- after about 1 minute (so the tracing captures correctly the actual state of the mts200) issue the service call or use the UI to set the hvac to 'auto'
- wait for the trace to finish and upload it here (you could re-configure the default tracing duration in the device config UI to be less than 10 minutes: say 3-4 so you have enough time to start it, wait a minute, and issue the command)

Also, in order to see if the preset modes mappings are correct you could manually operate the device from its panel (or the Meross app) and see if the correct preset/hvac modes are reported in HA. This should assure us meross_lan is correctly understanding the modes of operation

**Comment by john-wolff59** [comment]
hello @krahabb 
thanks a lot for you response.

I tried to do the default tracing.

And i write your what i have done :

1. Restart Home assistant with the MTS200 ON with the with the setpoint of 23 degrees.
2. MTS200 ON with the with the setpoint of 19 degrees via HA. => OK
3. MTS200 OFF via HA . => OK
4. MST200 On with the with the setpoint of 19 degrees via HA. => OK
5. Execution of my automation to put the MTS200 in manual mode (`set_temperature` at 24 degrees) via HA => OK
6. Execution of my automation to put the MTS200 in Auto mode (`set_hvac_mode: auto`) via HA => **NOK**
7. Execution of my automation to put the MTS200 in Auto mode (`set_hvac_mode: auto`) via HA => **NOK**
8. MTS200 ON with the with the setpoint of 17 degrees via meross apllication. => OK
9. Execution of my automation to put the MTS200 in Auto mode (`set_hvac_mode: auto`) via HA => **NOK**
10. MTS200 OFF via meross => OK
11. Execution of my automation to put the MTS200 in Auto mode (`set_hvac_mode: auto`) via HA => **NOK**
12. Execution of my automation to put the MTS200 in Auto mode (`set_hvac_mode: auto`) via HA => **NOK**

i attach the default tracing
I can make more test if you want.
Thanks
[config_entry-homekit_controller-7c27179fc852621445c00bc1e35c6c0d.json.txt](https://github.com/krahabb/meross_lan/files/11210154/config_entry-homekit_controller-7c27179fc852621445c00bc1e35c6c0d.json.txt)

**Reply by krahabb** [reply]
Hello!
This is a trace from the homekit_controller integration and if you're using that it could be a different story ;)
Maybe, when invoking the `set_hvac_mode` service you're using the device/entity through homekit_controller and the bug lies in that integration or even in the the 'whole' meross/homekit infrastructure (could be a bug in the homekit implementation of the device too)....

basically, in your system,  you could have two different devices/climates provided by homekit_controller and meross_lan which finally resemble to the same physical device but they could behave different since they are totally different paths/implementations.

**Reply by john-wolff59** [reply]
sorry i'm just starting...
I thought I had already gone through the Meross_LAN application.
Indeed, now I can control the heating in manual and/or automatic (schedule) mode.

Thank you so much @krahabb 
I close the topic
